src/tools/w3dviewer/graphicview.cpp

```diff
@@ -0,0 +1,1095 @@
+
+    if (!IsRendering) {
+        IsRendering = true;
+        CW3DViewDoc *document = reinterpret_cast<CW3DViewDoc *>(m_pDocument);
+
+        if (document->m_initialized && document->m_scene != nullptr && !m_renderingDisabled) {
+            unsigned int newtime = timeGetTime();
+            unsigned int currenttime = m_time;
+            m_time = newtime;
+            int tm = newtime - currenttime;
+            unsigned int sync;
+
+            if (time != 0) {
+                sync = time + W3D::Get_Sync_Time();
+            } else {
+                sync = tm * m_animationSpeed + W3D::Get_Sync_Time();
+            }
+
+            W3D::Sync(sync);
+
+            if (m_animationState == ANIMATION_PLAY && update) {
+                document->UpdateAnimation(tm * 0.001f * m_animationSpeed);
+            }
+
+            if (m_objectRotationFlags != 0 && update) {
+                UpdateObjectRotation();
+            }
+
+            if (m_lightRotationFlags != 0 && update) {
+                UpdateLightTransform();
+            }
+
+            if (document->m_model != nullptr && document->m_scene->Get_Auto_Switch_LOD()) {
+                ResetLODLevel(document->m_model);
+            }
+
+            W3D::Begin_Render(true, true, document->m_fogColor);
+            W3D::Render(document->m_textureScene, document->m_textureCamera);
+
+            if (document->m_backgroundObjectName.GetLength() > 0) {
+                W3D::Render(document->m_backgroundScene, document->m_docCamera);
+            }
+
+            unsigned long high = 0;
+            unsigned long clock = Get_CPU_Clock(high);
+            W3D::Render(document->m_scene, m_camera);
+            unsigned long clock2 = Get_CPU_Clock(high);
+            unsigned int frametime = (clock2 - clock) / 1000;
+            W3D::End_Render(true);
+
+            if (newtime - m_lastCountUpdate > 250) {
+                m_lastCountUpdate = newtime;
+                document->UpdateParticleCount();
+                int polys = 0;
+
+                if (document->m_model != nullptr) {
+                    polys = document->m_model->Get_Num_Polys();
+                }
+
+                reinterpret_cast<CMainFrame *>(AfxGetMainWnd())->UpdatePolyCount(polys);
+            }
+
+            reinterpret_cast<CMainFrame *>(AfxGetMainWnd())->UpdateStatusBar(frametime);
+        }
+
+        IsRendering = false;
+    }
+}
+
+void CGraphicView::UpdateCamera()
+{
+    int width;
+    int height;
+
+    if (m_isWindowed) {
+        RECT r;
+        GetClientRect(&r);
+        width = r.right - r.left;
+        height = r.bottom - r.top;
+    } else {
+        width = DeviceWidth;
+        height = DeviceHeight;
+    }
+
+    float hfov;
+    float vfov;
+
+    if (height <= width) {
+        hfov = 0.7853981852531433f;
+        vfov = (float)height / (float)width * 0.7853981852531433f;
+    } else {
+        hfov = (float)height / (float)width * 0.7853981852531433f;
+        vfov = 0.7853981852531433f;

```

Comment:
This value is `DEG_TO_RAD(45.f)`. Can make it a `constexpr` to type it just once.

Reply:
fixed.